packages/salesforcedx-vscode-core/test/jest/decorators/scratch-org-decorator.test.ts

```diff
@@ -0,0 +1,80 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import { mock } from 'sinon';
+import * as vscode from 'vscode';
+import URI from 'vscode-uri';
+import { monitorOrgConfigChanges, showOrg } from '../../../src/decorators';
+import { nls } from '../../../src/messages';
+
+describe('scratch org decorator', () => {
+  const testUser = 'test@username.com';
+  const browserIcon = `$(browser)`;
+  const openOrgCommand = 'sfdx.force.org.open';
+  const fakePathURI = URI.file('/Users/tester/mockNewFile.test.js');
+  const mockStatusBarItem = {
+    tooltip: '',
+    command: '',
+    show: jest.fn(),
+    text: ''
+  };
+  let onDidCreateEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  let onDidChangeEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  onDidCreateEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  onDidChangeEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  const mockWatcher = {
+    // onDidChange: onDidChangeEventEmitter.event,
+    // onDidCreate: onDidCreateEventEmitter.event
+    onDidChange: jest.fn(),
+    onDidCreate: jest.fn()
+  };
+  let mockFileSystemWatcher: jest.SpyInstance;
+  let mockGetDefaultUsername: jest.SpyInstance;
+  let mockCreateStatusBar: jest.SpyInstance;
+
+  beforeEach(() => {
+    mockCreateStatusBar = (vscode.window.createStatusBarItem as any).mockReturnValue(
+      mockStatusBarItem
+    );
+    mockFileSystemWatcher = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
+    mockGetDefaultUsername = jest
```

Comment:
nit: Could these be renamed to be consistent with the pattern we've followed in other recent tests:
_"methodName-Mock"_:
<img width="400" alt="2022-11-08_06-43-16" src="https://user-images.githubusercontent.com/46458081/200594857-56f83084-33a8-45fe-bee6-da0626e37383.png">